src/URI.js

```diff
@@ -1670,63 +1731,67 @@
     return this;
   };
   p.normalizePath = function(build) {
-    if (this._parts.urn) {
-      return this;
-    }
-
-    if (!this._parts.path || this._parts.path === '/') {
-      return this;
-    }
-
-    var _was_relative;
     var _path = this._parts.path;
-    var _leadingParents = '';
-    var _parent, _pos;
 
-    // handle relative paths
-    if (_path.charAt(0) !== '/') {
-      _was_relative = true;
-      _path = '/' + _path;
-    }
+    if (this._parts.urn) {
```

Comment:
It certainly can be; in order to do that, we'd have to duplicate the `this.build(!build)` line. That's probably not so bad, but I think I can just pull the parent-resolving part of the code into its own subroutine, and it won't be so hideously overindented then.